### PR TITLE
Plot: Save Dialog stops repeated calls to save

### LINKED ISSUE #26572: Plot: Save dialog does not close on saving
### Expected behavior
When saving a project, clicking the save button should close the dialog. 
### Actual behavior
The save dialog stays open and the save algorithm can be started multiple times. 
### Steps to reproduce the behavior
1. Load any file. 
2. Save the project.
3. The dialog should not close while the project is saving and so can be started multiple times. 
### Platforms affected
Tested on Ubuntu 18.04

### PR DESCRIPTION
**Pull request by cailafinn**
**Description of work.**
Save dialog now greys out the button and gives the user feedback that the project is saving, so they do not hit the button multiple times and queue multiple save operations. 

**To test:**
1. Load some data.
2. Save a project.
3. Check that the save button is greyed out while the project is saving. 

Fixes #26572 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by Harrietbrown** [comment]
It may well be worth being a seperate PR, but when saving, if you click the cancil button it will close the save project window but will not stop the saving process. If feasable could you make it so that closing the window also stops the loading?

**Comment by ConorMFinn** [comment]
> If feasable could you make it so that closing the window also stops the loading?

Due to the way that ProjectSave works, it makes multiple calls to SaveNexusPreocessed, and as sa result the whole save operation cannot be cancelled. 
I think the best solution would be to simply close the dialog after save is pressed, and allow the user to cancel from the Algorithm Progress Dialog should they wish to. Though I am open to suggestions as that isn't ideal.

**Comment by Harrietbrown** [comment]
In that case I would say disable the cancel button as well as the save while save project is working. they can always close it from the X button if they want it out of the way.

**Review by Harrietbrown — APPROVED** [review]
Good work 👍

File: MantidPlot/src/ProjectSaveView.cpp [hunk 1]
```diff
@@ -265,6 +265,10 @@ void ProjectSaveView::save(bool checked) {
   auto filePath = m_ui.projectPath->text();
   auto compress = filePath.endsWith(".gz");
 
+  m_ui.btnSave->setEnabled(false);
```
Comment (martyngigg) [inline MantidPlot/src/ProjectSaveView.cpp]: Is the dialog guaranteed to close after the save so the buttons will never be needed again?

Comment (martyngigg) [reply MantidPlot/src/ProjectSaveView.cpp]: @ConorMFinn Could I get clarification on this?

Comment (ConorMFinn) [reply MantidPlot/src/ProjectSaveView.cpp]: I don't believe so, if the file cannot be saved another dialog opens that closes the window once okay is clicked.

Comment (ConorMFinn) [reply MantidPlot/src/ProjectSaveView.cpp]: I don't believe so, if the file cannot be saved another dialog opens that closes the window once okay is clicked.

Comment (martyngigg) [reply MantidPlot/src/ProjectSaveView.cpp]: Thanks for clarifying.

**Pull request merged**